### ISSUE TITLE
Populating the patient synopsis panes via individual REST requests

### ISSUE THREAD

**Issue by robtweed** [PulseTile/PulseTile-React-Core#138]
When you select a patient in the PHR UI, the Home screen displays a set of 6 panels showing the latest 4 records for:

- Problems
- Contacts
- Allergies
- Medications
- Vaccinations
- Top 3 Things

Currently PulseTile asks the back-end for the data to populate these panels via a single REST API, eg:

GET /api/patients/9999999000

The problem is that this presents the back-end with a very large and quite time-consuming task, the details of which I won't bore you all with, but the main problem is that the middle tier cannot respond with the data for the panels until it has received, processed and assembled all the results for all 6 headings from all the back-end OpenEHR machines.

I'd like to propose that we switch to populating each panel separately via its own API, 

GET /api/patients/{patientId}/synopsis/{heading}

eg:

GET /api/patients/9999999000/synopsis/allergies

In other words, for PulseTile to fire off 6 requests to the middle tier asynchronously and at the same time, with each panel displaying its 4 records when the response is returned for its specific heading.  This will improve the perceived responsiveness of the UI at this stage of its data presentation, and also will allow the middle tier to distribute its load across multiple Node.js worker processes which should improve the overall time taken to retrieve and process all the headings.

The API heading names would be:

- problems
- contacts
- allergies
- medications
- vaccinations
- top3things

**Comment by robtweed** [comment]
One additional change related to this - another set of data returned by the original single API is the demographics data for the patient.  The UI will need to fetch this using another API:

GET /api/demographics/{{patientid}}

eg GET /api/demographics/9999999000

This returns the info to populate the patient banner (name, address, GP details etc)

**Comment by juliashulga** [comment]
task is done

**Comment by tony-shannon** [comment]
thanks @juliashulga 
Suggest we make sure this is working as requested soon, then we can consider closing this issue
T

**Comment by juliashulga** [comment]
**New adjustment:**
optimization in file src/components/pages/PatientsSummary/separate-selectors.js

Create one selector to replace it with current 6 selectors

**Comment by tony-shannon** [comment]
thanks Julia
That adjustment remains TODO

**Comment by tony-shannon** [comment]
@juliashulga  please check with @robtweed  that he is happy this can be closed

**Comment by juliashulga** [comment]
Hi @robtweed can this task be closed now ? is everything ok here ?

**Comment by robtweed** [comment]
Yes it seems to work OK now
